core/controllers/acl_decorators_test.py

```diff
@@ -5175,6 +5204,38 @@ def test_user_cannot_access_story_when_topic_is_not_published(self) -> None:
                 % self.story_url_fragment,
                 expected_status_int=404)
 
+    def test_guest_cannot_access_story(self) -> None:
+        self.logout()
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.mock_testapp.get(
+                '/mock_story_data/staging/topic/%s' % self.story_url_fragment,
+                expect_errors=True)
+            self.assertEqual(response.status_int, 401)
+
+    def test_user_cannot_access_story_with_empty_topic_id(self) -> None:
```

Comment:
How do you know that the error here is due to the invalid story ID rather than the empty topic ID?